x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/DefaultAuthenticationFailureHandler.java

```diff
@@ -100,12 +101,20 @@ public ElasticsearchSecurityException failedAuthentication(TransportMessage mess
 
     @Override
     public ElasticsearchSecurityException exceptionProcessingRequest(RestRequest request, Exception e, ThreadContext context) {
+        // a couple of authn processing errors can also return 500 & 503, besides the obvious 401
+        if (e instanceof ElasticsearchAuthenticationProcessingError) {
+            return (ElasticsearchAuthenticationProcessingError) e;
+        }
         return createAuthenticationError("error attempting to authenticate request", e, (Object[]) null);
```

Comment:
Only raising exceptions with a specific type (`ElasticsearchAuthenticationProcessingError`) AND while invoking the request processing error handler (and not any other handler), result in 500 or 503; anything else goes to 401 .